### PR TITLE
SkipList object

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: src/store/index/mod.rs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+mod skip_index;
+mod skip_index_builder;
+
+use crate::DocId;
+
+pub use self::skip_index::SkipIndex;
+pub use self::skip_index_builder::SkipIndexBuilder;
+
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub struct Checkpoint {
```
Comment (fulmicoton) [inline src/store/index/mod.rs:22]: comment about [start_doc..end_doc)

File: src/store/index/block.rs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+use crate::DocId;
+use std::io;
+
+/// Represents a block of checkpoints.
+///
+/// The DocStore index checkpoints are organized into block
+/// for code-readability and compression purpose.
+///
+/// A block can be of any size.
+pub struct Block {
```
Comment (fulmicoton) [inline src/store/index/block.rs]: Rename

File: src/store/index/mod.rs (hunk 1 above)
Comment (fulmicoton) [inline src/store/index/mod.rs:22]: move into block.rs and rename checkpoint.rs

File: src/store/index/block.rs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    pub checkpoints: Vec<Checkpoint>,
+}
+
+impl Default for Block {
+    fn default() -> Block {
+        Block {
+            checkpoints: Vec::with_capacity(2 * PERIOD),
```
Comment (fulmicoton) [inline src/store/index/block.rs]: rename

File: src/store/reader.rs [hunk 4]
```diff
@@ -1,67 +1,67 @@
...
 
     pub(crate) fn block_data(&self) -> io::Result<OwnedBytes> {
         self.data.read_bytes()
     }
 
-    fn compressed_block(&self, addr: usize) -> io::Result<OwnedBytes> {
-        let (block_len_bytes, block_body) = self.data.slice_from(addr).split(4);
-        let block_len = u32::deserialize(&mut block_len_bytes.read_bytes()?)?;
-        block_body.slice_to(block_len as usize).read_bytes()
+    fn compressed_block(&self, start_offset: u64, end_offset: u64) -> io::Result<OwnedBytes> {
```
Comment (fulmicoton) [inline src/store/reader.rs]: return a slice

File: src/store/reader.rs [hunk 5]
```diff
@@ -74,11 +74,13 @@ impl StoreReader {
     /// It should not be called to score documents
     /// for instance.
     pub fn get(&self, doc_id: DocId) -> crate::Result<Document> {
-        let (first_doc_id, block_offset) = self.block_offset(doc_id);
-        self.read_block(block_offset as usize)?;
+        let checkpoint = self.block_offset(doc_id).ok_or_else(|| {
```
Comment (fulmicoton) [inline src/store/reader.rs]: rename  block_checkpoint

**Review by guilload — COMMENTED** [review]
Just left two nit comments.

**Pull request merged**